src/ExperimentalFunds.mo

```diff
@@ -0,0 +1,54 @@
+/// Funds and units
+///
+/// Provides imperative operations for observing funds, transferring funds and
+/// observing refunds of various units.
+/// The two currently support units are `#cycle` and `#icpt`, measuring
+/// computational cost and Internet Computer tokens, respectively.
+
+/// This low-level API is experimental and likely to change, including the
+/// addition of other units and alternative syntactic support for
+/// manipulating funds.
+///
+/// Note: since unit `#cycle` is used to measure computation, the value of
+/// `balance(#cycle)` generally decreases from one call to the next.
+
+import Prim "mo:prim";
+module {
+
+  /// Units for funds: `{#cycle; #icpt}`.
+  public type Unit = Prim.Unit;
+
+  /// Returns the actor's current balance of unit `u` as `amount`.
+  public let balance : (u : Unit) -> (amount : Nat64) = Prim.fundsBalance;
```

Comment:
Yes, this is low-level, but I’d at least convert to `Nat` for the user here, like we do for time. Presumably the system api will support bignum (or just bigger num, i.e. i128, as suggested by Johan) at some point in the future.

Reply:
Ok, but just for balance or the other amounts too? It a bit weird that we dev_set_funds sets the balance using a Nat, but we can only read as a Nat64 in the API.

Reply:
Yes, maybe we should have done the System API in terms of BigNums right away…